### PR TITLE
Add tracking for backup storage info bar chart

### PR DESCRIPTION
**Pull request by manzoorwanijk**
#### Changes proposed in this Pull Request:

* This PR adds tracking for the new bar chart added in #23374 

#### Jetpack product discussion
HACK WEEK project p6TEKc-61M-p2

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Boot up the branch and run a JT site and connect Jetpack with Free plan
* Goto At a Glance page. 
* Ensure that you have more than 20 posts published on the site OR you can also manually change (via React Dev tools) the `posts` prop to more than 20 in `BackupUpdare` component added in #23374 
* After the BarChart is visible, confirm that you see in Network tab new tracking request with event name `jetpack_wpa_aag_backup_bar_chart_view`
* Close the popup that blocks the bar chart
* Confirm that you see in Network tab new tracking request with event name being `jetpack_wpa_click`, `target` and `type` being `backup_bar_chart_close` and `dismiss` respectively.

### CODE REVIEW

File: projects/plugins/jetpack/_inc/client/at-a-glance/backup-upgrade/index.jsx [hunk 1]
```diff
@@ -23,8 +24,18 @@ const BackupUpgrade = ( { comments, isFetchingData, plugins, posts } ) => {
 	const [ showPopup, setShowPopup ] = useState( true );
 
 	const onClosePopup = useCallback( () => {
+		const clickEventProps = {
+			comments,
+			plugins,
+			posts,
+			target: 'backup_bar_chart_close',
+			type: 'dismiss',
+		};
```
Comment (elliottprogrammer) [inline projects/plugins/jetpack/_inc/client/at-a-glance/backup-upgrade/index.jsx:33]: I'm not exactly sure about the naming properties of `recordJetpackClick()`, ie- { `target`, `feature`, `type` } and what they should represent exactly, maybe someone from Crew can verify if your `target` and `type` properties are following convention properly.

**Review by elliottprogrammer — DISMISSED** [review]
Tested, and the tracking events are firing correctly. The code LGTM, just a minor comment below to make sure we're naming the tracking properties properly.
Otherwise, everything LGTM! 👍

File: projects/plugins/jetpack/_inc/client/at-a-glance/backup-upgrade/index.jsx (hunk 1 above)
Comment (manzoorwanijk) [reply projects/plugins/jetpack/_inc/client/at-a-glance/backup-upgrade/index.jsx:33]: CC @Automattic/jetpack-crew

Comment (jeherve) [reply projects/plugins/jetpack/_inc/client/at-a-glance/backup-upgrade/index.jsx:33]: > Crew can verify if your target and type properties are following convention properly.

This looks good to me. 👍

**Review by jeherve — COMMENTED** [review]
This looks good, but seems to be triggered too often in my tests:

https://videopress.com/v/sMyXVKu3

**Comment by manzoorwanijk** [comment]
> This looks good, but seems to be triggered too often in my tests:

I have added a fix.

It is happening because of `<QueryStatsData />` rendered in `<DashStats />` in "At a Glance" page. When you switch back to the tab, the query for stats data runs again, which triggers re-render.

**Review by jeherve — APPROVED** [review]
This tests well for me, should be good to merge!

Upon looking at that banner again, I would have 3 suggestions for improvements, in future PRs:

- Use `numberFormat` to format the numbers displayed in the banner.
- When dismissing the banner, dismiss the whole banner, including `JetpackBanner`
- Remember the dismissal so the banner doesn't come back if I navigate away from that view and come back to it a minute later.

**Comment by manzoorwanijk** [comment]
> I would have 3 suggestions for improvements, in future PRs:

Thank you. Will try to implement those suggestion in an upcoming PRs.

**Pull request merged**